### ISSUE TITLE
Document Calico's use of upstream CNI plugins

### ISSUE THREAD

**Issue by caseydavenport** [projectcalico/calico#6004]
Calico installs the upstream containernetworking/plugins binaries as part of it's install-cni container.

We should document this, explain the reasons for it, and how to control it.

## Possible Solution

Update Calico the Hardway, as well as the CNI reference documentation to explain this behavior.

Possibly, add new controls for this in the operator or remove the binaries entirely from our install-cni container. 

## Context

Slack thread: https://calicousers.slack.com/archives/CPTH1KS00/p1651514474421769

**Comment by dlipovetsky** [comment]
I just looked at the ["hard way" doc](https://github.com/projectcalico/calico/blob/2b3469b/calico/getting-started/kubernetes/hardway/install-node.md) again, and I'm not sure if it needs to be changed.

The DaemonSet manifest in this doc has no initContainers. But the CNI is [installed using an initContainer](https://github.com/projectcalico/calico//blob/4a0fd9684d172e40c3c97f40c42c51ab5b92f708/manifests/calico.yaml#L4391-L4396).

Does the manifest on the doc need to be updated? Or is it fine as-is, and should some other doc be updated? :thinking:

**Comment by caseydavenport** [comment]
Yep, you're correct - the hardway documentation doesn't use our init container to install the CNI plugin, but rather it has the user install it by hand in this doc: https://github.com/projectcalico/calico/blob/2b3469b257d3306432ac274db5193693186afb3b/calico/getting-started/kubernetes/hardway/install-cni-plugin.md

We still might want something in the operator to allow configuration of this, though.

**Comment by VannTen** [comment]
@caseydavenport 

> We should document this, explain the reasons for it, and how to control it.

The slack thread appears to be gone, unfortunately. Would you have any information on why it's bundled this way ? (or maybe that just happened ?)

**Comment by caseydavenport** [comment]
There's a bit of historical context for this - Calico started bundling the CNI plugins as part of its solution before there was an official package for installing them! 

But there are also more modern reasons. For example, we maintain our own versions of these plugins to ensure they are up-to-date with the latest CVE fixes from the Go standard libraries, etc.
